kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/http/AbstractSimultaneousConnectionsTest.java

```diff
@@ -94,29 +101,22 @@ private void withHttp1() throws IOException {
   }
 
   @Test
-  @DisplayName("Should be able to make 1024 simultaneous HTTP/1.x connections before processing the response")
+  @DisplayName("Should be able to make 2048 simultaneous HTTP/1.x connections before processing the response")
   @DisabledOnOs(OS.WINDOWS)
   public void http1Connections() throws Exception {
-    final CountDownLatch activeConnections = new CountDownLatch(MAX_HTTP_1_CONNECTIONS);
-    httpServer.createContext("/http", exchange -> {
-      activeConnections.countDown();
-      try {
-        activeConnections.await(20, TimeUnit.SECONDS);
-        exchange.sendResponseHeaders(204, -1);
-      } catch (Exception ignore) {
-        // NO OP
-      }
-    });
+    final CyclicBarrier barrier = new CyclicBarrier(2);
+    httpServer.createContext("/http",
+        new DelayedResponseHandler(MAX_HTTP_1_CONNECTIONS, barrier, exchange -> exchange.sendResponseHeaders(204, -1)));
     try (final HttpClient client = clientBuilder.build()) {
       final Collection<CompletableFuture<HttpResponse<AsyncBody>>> asyncResponses = ConcurrentHashMap.newKeySet();
       final HttpRequest request = client.newHttpRequestBuilder()
           .uri(String.format("http://localhost:%s/http", httpServer.getAddress().getPort()))
           .build();
       for (int it = 0; it < MAX_HTTP_1_CONNECTIONS; it++) {
         asyncResponses.add(client.consumeBytes(request, (value, asyncBody) -> asyncBody.consume()));
+        barrier.await(1, TimeUnit.SECONDS);
       }
-      assertThat(activeConnections.await(20, TimeUnit.SECONDS)).isTrue();
-      CompletableFuture.allOf(asyncResponses.toArray(new CompletableFuture[0])).get(10, TimeUnit.SECONDS);
+      CompletableFuture.allOf(asyncResponses.toArray(new CompletableFuture[0])).get(20, TimeUnit.SECONDS);
```

Comment:
The 20 second value seemed problematic in some of my testing with the non-okhttp clients, thus the change to how the other pr was awaiting this and the similar timeouts.  The improved thread management made things better, but you may want to consider increasing this for github runs just to aliviate the possibility of spurious failures.